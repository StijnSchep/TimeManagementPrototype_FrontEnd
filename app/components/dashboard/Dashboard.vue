<template>
  <Page actionBarHidden="true">
    <BottomNavigation>
      <TabStrip style="border-top: 1px solid black;">
        <TabStripItem>
          <Label col="1" class="fas" :text="'fa-home' | fonticon" />
        </TabStripItem>
        <TabStripItem>
          <Label col="1" class="fa" :text="'fa-clock' | fonticon" />
        </TabStripItem>
        <TabStripItem>
          <Label col="1" class="fa" :text="'fa-calendar-alt' | fonticon" />
        </TabStripItem>
        <TabStripItem>
          <Label col="1" class="fas" :text="'fa-user-circle' | fonticon" />
        </TabStripItem>
        <TabStripItem>
          <Label col="1" class="fas" :text="'fa-cog' | fonticon" />
        </TabStripItem>
      </TabStrip>

      <TabContentItem>
        <Frame>
          <HomeComponent :token="token" />
        </Frame>
      </TabContentItem>

      <TabContentItem>
        <Frame>
          <HoursComponent :token="token" />
        </Frame>
      </TabContentItem>

      <TabContentItem>
        <Frame>
          <CalendarComponent :token="token" />
        </Frame>
      </TabContentItem>

      <TabContentItem>
        <Frame>
          <AccountComponent :token="token" />
        </Frame>
      </TabContentItem>

      <TabContentItem>
        <Frame>
          <SettingComponent :token="token" />
        </Frame>
      </TabContentItem>
    </BottomNavigation>
  </Page>
</template>

<script>
import AccountComponent from "./AccountPage";
import CalendarComponent from "./CalendarPage";
import HoursComponent from "./HoursPage";
import HomeComponent from "./HomePage";
import SettingComponent from "./SettingPage";
import * as application from "tns-core-modules/application";

export default {
  components: {
    HomeComponent,
    HoursComponent,
    CalendarComponent,
    AccountComponent,
    SettingComponent
  },
  created() {
    this.$store.dispatch("checkIn/fetchAllWorkedHours", this.token);
  },
  computed: {
    token() {
      return this.$store.state.auth.token;
    }
  }
};

if ( application.android ) {
  application.android.on(
          application.AndroidApplication.activityBackPressedEvent,
          args => {
            args.cancel = true;
          }
  );
}
</script>

<style scoped>
TabStripItem {
  font-size: 25px;
  color: #dddddd;
}
TabStripItem:active {
  color: #00a0d1;
}
</style>
