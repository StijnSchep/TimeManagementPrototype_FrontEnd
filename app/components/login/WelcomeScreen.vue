<template>
    <Page class="body" actionBarHidden="true">
        <FlexboxLayout class="page">
            <Label :text="'Welkom ' + employee.firstname + ' ' + employee.lastname" class="welcome-txt"/>
            <ActivityIndicator :busy="busy" color="#fff" @busyChange="onBusyChanged" />
        </FlexboxLayout>
    </Page>
</template>

<script>
    import Dashboard from "../dashboard/Dashboard";

    export default {
        props: ['employee', 'token'],
        data() {
            return {
                busy: true
            }
        },
        methods: {
            onBusyChanged() {
                this.goToDashboard()
            },
            goToDashboard() {
                this.$navigateTo(Dashboard, {
                    clearHistory: true,
                    transition: {
                        name: 'fade',
                        duration: 400
                    },
                    props: {
                        token: this.token
                    }
                })
            },
            sleep(self) {
                setTimeout(() => {
                    self.busy = false;
                }, 2000)
            }
        },
        created() {
            this.sleep(this);
        }
    }
</script>

<style scoped>
    .body {
        background-image: url("~/assets/images/SplashScreen.jpg");
        background-repeat: no-repeat;
        background-position: center top;
        background-size: cover;
    }

    .page {
        align-items: center;
        vertical-align: center;
        flex-direction: column;
    }

    .welcome-txt {
        text-align: center;
        color: white;
        font-size: 30px;
        margin-bottom: 10px;
    }
</style>
