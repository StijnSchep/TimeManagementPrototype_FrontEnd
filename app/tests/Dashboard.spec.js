import { shallowMount, createLocalVue } from '@vue/test-utils';
import Dashboard from '../components/dashboard/Dashboard';
import Vuex from 'vuex'
import WelcomeScreen from "../components/login/WelcomeScreen";
import { getString } from 'tns-core-modules/application-settings';

jest.mock('tns-core-modules/application-settings', () => ({
  getString: jest.fn(() => "")
}));

jest.mock('nativescript-fingerprint-auth', () => ({
  FingerprintAuth: jest.fn(),
  getString: jest.fn(() => "")
}));

jest.mock('tns-core-modules/application', () => ({
  android: {
    on: jest.fn()
  },
  AndroidApplication: {
    activityBackPressedEvent: {}
  }
}));

jest.mock("tns-core-modules/ui/dialogs", () => ({
  alert: jest.fn(() => Promise.resolve())
}));

const localVue = createLocalVue();
localVue.use(Vuex)

describe('Dashboard.vue', () => {
  let store;
  let state;
  let actions;

  beforeEach(() => {
    state = {
      token: ''
    };
    actions = {
      getToken: jest.fn()
    };
    store = new Vuex.Store({
      modules: {
        auth: {
          state,
          actions
        }
      }
    });
  });

  // jest.mock('nativescript-drop-drown');



  it('mounts and renders', async () => {
    const wrapper = shallowMount(Dashboard, { store, localVue });
    await expect(wrapper.html()).toBeTruthy();
  });

  test('renders correctly', async () => {
    const wrapper = shallowMount(Dashboard, { store, localVue });
    await expect(wrapper.element).toMatchSnapshot();
  });

  it('is a Vue instance', async () => {
    const wrapper = shallowMount(Dashboard, { store, localVue });
    await expect(wrapper.isVueInstance()).toBeTruthy();
  });

  it('Expect component to exist', async () => {
    const wrapper = shallowMount(Dashboard, { store, localVue });
    await expect(wrapper).toBeTruthy();
  });

  it('Expect component to have Page element', async () => {
    const wrapper = shallowMount(Dashboard, { store, localVue });
    await expect(wrapper.find('Page')).toBeTruthy();
  });

  it('Expect component to have StackLayout element', async () => {
    const wrapper = shallowMount(Dashboard, { store, localVue });
    await expect(wrapper.find('StackLayout')).toBeTruthy();
  });

  it('Expect component to have template element', async () => {
    const wrapper = shallowMount(Dashboard, { store, localVue });
    await expect(wrapper.find('template')).toBeTruthy();
  });

  it('Expect component to have script element', async () => {
    const wrapper = shallowMount(Dashboard, { store, localVue });
    await expect(wrapper.find('script')).toBeTruthy();
  });

  it('Expect component to have style element', async () => {
    const wrapper = shallowMount(Dashboard, { store, localVue });
    await expect(wrapper.find('style')).toBeTruthy();
  });

  it('Expect component to have Label element', async () => {
    const wrapper = shallowMount(Dashboard, { store, localVue });
    await expect(wrapper.find('Label')).toBeTruthy();
  });

  it('Expect component to have BottomNavigation element', async () => {
    const wrapper = shallowMount(Dashboard, { store, localVue });
    await expect(wrapper.find('BottomNavigation')).toBeTruthy();
  });

  it('Expect component to have TabStrip element', async () => {
    const wrapper = shallowMount(Dashboard, { store, localVue });
    await expect(wrapper.find('TabStrip')).toBeTruthy();
  });

  it('Expect component to have TabStripItem element', async () => {
    const wrapper = shallowMount(Dashboard, { store, localVue });
    await expect(wrapper.find('TabStripItem')).toBeTruthy();
  });

  it('Expect component to have TabContentItem element', async () => {
    const wrapper = shallowMount(Dashboard, { store, localVue });
    await expect(wrapper.find('TabContentItem')).toBeTruthy();
  });

  it('Expect component to have Frame element', async () => {
    const wrapper = shallowMount(Dashboard, { store, localVue });
    await expect(wrapper.find('Frame')).toBeTruthy();
  });

  it('Expect component to have HomeComponent element', async () => {
    const wrapper = shallowMount(Dashboard, { store, localVue });
    await expect(wrapper.find('HomeComponent')).toBeTruthy();
  });

  it('Expect component to have HoursComponent element', async () => {
    const wrapper = shallowMount(Dashboard, { store, localVue });
    await expect(wrapper.find('HoursComponent')).toBeTruthy();
  });

  it('Expect component to have CalendarComponent element', async () => {
    const wrapper = shallowMount(Dashboard, { store, localVue });
    await expect(wrapper.find('CalendarComponent')).toBeTruthy();
  });

  it('Expect component to have AccountComponent element', async () => {
    const wrapper = shallowMount(Dashboard, { store, localVue });
    await expect(wrapper.find('AccountComponent')).toBeTruthy();
  });
});
