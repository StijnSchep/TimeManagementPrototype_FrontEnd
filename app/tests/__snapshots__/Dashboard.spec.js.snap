// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard.vue renders correctly 1`] = `
<page-stub
  actionbarhidden="true"
>
  <bottomnavigation-stub>
    <tabstrip-stub
      style="border-top: 1px solid black;"
    >
      <tabstripitem-stub>
        <label
          class="fas"
          col="1"
          text="fa-home"
        />
      </tabstripitem-stub>
       
      <tabstripitem-stub>
        <label
          class="fa"
          col="1"
          text="fa-clock"
        />
      </tabstripitem-stub>
       
      <tabstripitem-stub>
        <label
          class="fa"
          col="1"
          text="fa-calendar-alt"
        />
      </tabstripitem-stub>
       
      <tabstripitem-stub>
        <label
          class="fas"
          col="1"
          text="fa-user-circle"
        />
      </tabstripitem-stub>
       
      <tabstripitem-stub>
        <label
          class="fas"
          col="1"
          text="fa-cog"
        />
      </tabstripitem-stub>
    </tabstrip-stub>
     
    <tabcontentitem-stub>
      <frame-stub>
        <homecomponent-stub
          token=""
        />
      </frame-stub>
    </tabcontentitem-stub>
     
    <tabcontentitem-stub>
      <frame-stub>
        <hourscomponent-stub
          token=""
        />
      </frame-stub>
    </tabcontentitem-stub>
     
    <tabcontentitem-stub>
      <frame-stub>
        <calendarcomponent-stub
          token=""
        />
      </frame-stub>
    </tabcontentitem-stub>
     
    <tabcontentitem-stub>
      <frame-stub>
        <accountcomponent-stub
          token=""
        />
      </frame-stub>
    </tabcontentitem-stub>
     
    <tabcontentitem-stub>
      <frame-stub>
        <settingcomponent-stub
          token=""
        />
      </frame-stub>
    </tabcontentitem-stub>
  </bottomnavigation-stub>
</page-stub>
`;
